pkg/liqo-controller-manager/shadowpod-controller/shadowpod_controller.go

```diff
@@ -41,6 +43,13 @@ type Reconciler struct {
 	Scheme *runtime.Scheme
 }
 
+func podShouldBeUpdated(newPod *vkv1alpha1.ShadowPod, curPod *corev1.Pod) bool {
```

Comment:
```suggestion
func podShouldBeUpdated(newObj, oldObj client.Object) bool {
```

Could be generalized to something like this, to also use it for the predicate.